### PR TITLE
Added warnings for invalid fields that are required, and changed all itemId's to batchId

### PR DESCRIPTION
**Pull request by zainnk902**
### Administrative Info
Monday Board ID: 853628905

### Changes
What changes did you make?
- Changed all itemId's to batchId
- Made itemName and batchId required to add a new item
- gave warnings for invalid fields

### Testing
How did you confirm your changes worked? 
- Tested by entering empty fields for the itemName and batchId, and gave negative or zero weight to the batches to make sure they were flagged as invalid

### Confirmation of Change 

| --- |
Screen Shot 2021-02-22 at 6 20 44 PM

Entering any invalid fields for the itemName, batchId, or wight will give a warning, and you won't be able to add teh item until all warnings are gone.

### CODE REVIEW

File: client/src/home/add_button/AddButton.js [hunk 1]
```diff
@@ -47,23 +49,44 @@ function AddItemForm(props) {
...
           <Form.Group>
-            <Form.Label>Item Number</Form.Label>
+            <Form.Label>Batch ID</Form.Label>
             <Form.Control
               type="text"
               placeholder="e.g. 1234"
               onChange={(event) => {
-                setItemId(event.target.value.trim().toLowerCase());
+                let val = event.target.value.trim().toLowerCase();
+                if(val.length > 0){
```
Comment (hahsu) [inline client/src/home/add_button/AddButton.js]: For now at least, the client wanted a 4 digit number, can we change this to be val.length === 4, and then on line 88 the warning can be something like Four digit Batch ID is required

**Review by hahsu — CHANGES_REQUESTED** [review]

File: client/src/home/add_button/AddButton.js [hunk 2]
```diff
@@ -47,23 +49,44 @@ function AddItemForm(props) {
             <Form.Control
               type="text"
               placeholder="e.g. Apples"
-              onChange={(event) =>
-                setItemName(event.target.value.trim().toLowerCase())
-              }
+              onChange={(event) =>{
+                let val = event.target.value.trim().toLowerCase();
+                if(val.length > 0){
+                  setItemName(event.target.value.trim().toLowerCase());
```
Comment (AnsonLee99) [inline client/src/home/add_button/AddButton.js]: The item name should be set regardless of the field's length. The itemName state should always reflect what is in the field, even if the  field is empty.

File: client/src/home/add_button/AddButton.js [hunk 3]
```diff
@@ -47,23 +49,44 @@ function AddItemForm(props) {
...
-            <Form.Label>Item Number</Form.Label>
+            <Form.Label>Batch ID</Form.Label>
             <Form.Control
               type="text"
               placeholder="e.g. 1234"
               onChange={(event) => {
-                setItemId(event.target.value.trim().toLowerCase());
+                let val = event.target.value.trim().toLowerCase();
+                if(val.length > 0){
+                  setbatchId(event.target.value.trim().toLowerCase());
```
Comment (AnsonLee99) [inline client/src/home/add_button/AddButton.js]: Same issue here. The batchID state should always reflect what is currently in the field even if the field is empty.

File: client/src/home/add_button/AddButton.js [hunk 4]
```diff
@@ -28,13 +28,15 @@ function AddButton() {
 function AddItemForm(props) {
   // Input fields
   const [itemName, setItemName] = useState("");
-  const [itemId, setItemId] = useState("");
+  const [batchId, setbatchId] = useState("");
```
Comment (AnsonLee99) [inline client/src/home/add_button/AddButton.js]: Let's keep the capitalization consistent. "setbatchId" --> "setBatchId"

File: client/src/home/add_button/AddButton.js [hunk 5]
```diff
@@ -127,10 +150,19 @@ function AddItemForm(props) {
...
-            props.closeForm();
-          }}
+          
+            onClick={() => {
+              if(itemName.length > 0 && batchId.length > 0){
+                postNewItem(cookies.token, itemName, batchId, weight, outByDate);
+                props.closeForm();
+              }
+              else{
+                <Button.Text className="warning">One or more required feild is invalid</Button.Text>
```
Comment (AnsonLee99) [inline client/src/home/add_button/AddButton.js]: Does this show which fields are invalid?

**Review by AnsonLee99 — CHANGES_REQUESTED** [review]
Few minor changes. Left comments in the "Files changed" tab.

File: client/src/home/add_button/AddButton.js (hunk 5 above)
Comment (zainnk902) [reply client/src/home/add_button/AddButton.js]: Ah my bad I thought I took that out. That was an error message if someone tried to click the submit button and didn't have a required field, but that's not needed anymore

File: client/src/home/add_button/AddButton.js (hunk 1 above)
Comment (zainnk902) [reply client/src/home/add_button/AddButton.js]: Fixed!

File: client/src/home/add_button/AddButton.js (hunk 3 above)
Comment (zainnk902) [reply client/src/home/add_button/AddButton.js]: Fixed!

File: client/src/home/add_button/AddButton.js (hunk 2 above)
Comment (zainnk902) [reply client/src/home/add_button/AddButton.js]: Fixed!

**Review by AnsonLee99 — APPROVED** [review]
lgtm!

**Review by hahsu — APPROVED** [review]
lgtm, thanks!!

File: client/src/home/add_button/AddButton.js [hunk 6]
```diff
@@ -29,13 +29,15 @@ function AddButton(props) {
...
-  const [itemId, setItemId] = useState("");
+  const [batchId, setBatchId] = useState("");
   const [lbPerHouse, setLbPerHouse] = useState(0);
   const [weight, setWeight] = useState(0);
   const [outByDate, setOutByDate] = useState(new Date());
 
   const [cookies] = useCookies(["token"]);
   const [showWeightWarning, setWeightWarning] = useState(false);
+  const [showitemNameWarning, setitemNameWarning] = useState(false);
+  const [showbatchIdWarning, setbatchIdWarning] = useState(false);
```
Comment (petabite) [inline client/src/home/add_button/AddButton.js:40]: Kind of a nitpick: maybe want to camel case `showitemNameWarning` and `showbatchIdWarning`

**Review by petabite — APPROVED** [review]
Works as expected locally for me. Just one nitpick in the comments. After you fix the issue on line 202, you're all good!

**Pull request merged**